### ISSUE TITLE
`incompatible_msrv` wording is confusing

### ISSUE THREAD

**Issue by xobs** [rust-lang/rust-clippy#16995]
### Summary

The warning that appears with `incompatible_msrv` [is confusing](https://github.com/rust-lang/rust-clippy/discussions/16994).

The actual text is:

```
warning: current MSRV (Minimum Supported Rust Version) is `1.59.0` but this item is stable since `1.64.0`
```

It should read:

```
warning: current MSRV (Minimum Supported Rust Version) is `1.59.0` but this item comes from a crate with the a MSRV of `1.64.0`
```

### Reproducer

Code:

Create a new project with `--edition 2021` and add `rust-version = "1.59.0"` to `Cargo.toml`

```rust
fn main() {
    println!("c_int: {}", core::mem::size_of::<libc::c_int>());
}
```

Current output:

```
warning: current MSRV (Minimum Supported Rust Version) is `1.59.0` but this item is stable since `1.64.0`
```

Desired output:
```
warning: current MSRV (Minimum Supported Rust Version) is `1.59.0` but this item comes from a crate with the a MSRV of `1.64.0`
```

### Version

```text
rustc 1.95.0 (59807616e 2026-04-14)
binary: rustc
commit-hash: 59807616e1fa2540724bfbac14d7976d7e4a3860
commit-date: 2026-04-14
host: x86_64-unknown-linux-gnu
release: 1.95.0
LLVM version: 22.1.2
```

**Comment by samueltardieu** [comment]
The original message is correct: `libc` is reexporting `c_int` from `core::ffi::c_int` (through a `pub use`), and `core::ffi::c_int` is stable since Rust 1.64.0. You would get the same message if you used `core::ffi::c_int` directly.

**Comment by xobs** [comment]
I see. I'm sorry for the noise then. I thought the issue was that [the MSRV of the crate is 1.65.0](https://github.com/rust-lang/libc/blob/41f2d1d2cc557e0ff46a1dfc3225ac1e643fbf99/Cargo.toml#L12) and the message was just wrong.

Shouldn't it give a warning due to the MSRV of the crate, though? Or is that a separate lint?

**Comment by samueltardieu** [comment]
> Shouldn't it give a warning due to the MSRV of the crate, though? Or is that a separate lint?

I would think that would be the job of Cargo to warn about incompatible MSRV (and I think recent Cargo resolvers try to prevent this from happening in the first place).

**Comment by xobs** [comment]
Thanks for confirming that.
